### DISCUSSION TITLE
Remove the spell cost from the Shrines and Pyramid windows

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4617] [Ideas]
Since the spell cost has been removed from the Mage Guild screen, I suggest removing the spell cost from the Shrines and Pyramid windows as well.

Also, the player should be able to right click on the spell graphic in the Shrines and Pyramid windows to display the spell information window.

In fheroes2:

![Spell cost 001](https://user-images.githubusercontent.com/43583994/141351741-f7f7686a-85e4-4861-8ef7-502751de07b7.png)

![Spell cost 002](https://user-images.githubusercontent.com/43583994/141351755-0facd375-203c-41ed-a9d8-03a50c8e9db0.png)

![Spell cost 003](https://user-images.githubusercontent.com/43583994/141351771-6a7e6e7a-c93f-49d4-8023-11b57d970012.png)

![Spell cost 004](https://user-images.githubusercontent.com/43583994/141351788-97e7db1a-fea2-4ead-a889-617c715ecada.png)

![Pyramid message 02](https://user-images.githubusercontent.com/43583994/141346391-2b8561d6-b7b1-43c0-b882-88490205b7c6.png)

**Comment by Branikolog** [comment]
"+" from my side for this enhancement.
Even "**++**"

**Reply by Branikolog** [reply]
I think in spell description during battle spell cost should be removed either
![Spell description](https://user-images.githubusercontent.com/55348946/141685964-9d9f26a0-7839-492b-8069-3635f4dc7b44.png)

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> I think in spell description during battle spell cost should be removed either
> ![Spell description](https://user-images.githubusercontent.com/55348946/141685964-9d9f26a0-7839-492b-8069-3635f4dc7b44.png)

In this case it is better to leave the cost of the spell, this information is useful when the player consults the number of remaining spell points of the hero (for the opponent's hero and even for the player's hero) by right clicking on the hero.

**Reply by Branikolog** [reply]
> the player consults the number of remaining spell points of the hero

What is the point to know here spell cost, while player will see it during consulting a book?

I think, spell cost while checking opponent's spell - is a secret information for a player. :D Should hero know such details, for example, of new spell, which he never seen before?

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > the player consults the number of remaining spell points of the hero
> 
> What is the point to know here spell cost, while player will see it during consulting a book?

The player can see the spell information right there, without having to open the hero's screen and the Magic Book window. :)

Also notice that this window is the spell information window, not a window that indicates that the hero has obtained a spell (and that, in this window, the player should be able to right-click on the spell graphic to display... the spell information window).
 
> I think, spell cost while checking opponent's spell - is a secret information for a player. :D Should hero know such details, for example, of new spell, which he never seen before?

For the spells of the opposing hero, the spell information should be available (even for a new spell).
The player can deduce the cost of the opponent's spells by consulting the remaining spell points of the opponent's hero before and after casting the spell, so there is nothing secret here. :)

**Comment by LeHerosInconnu** [comment]
It's okay the way it works currently.
I close the discussion.
